### PR TITLE
Tooling Upgrade

### PR DESCRIPTION
**Pull request by jacobwgillespie**
**Work In Progress**

This adds some tooling upgrades from @svg-icons:

* An automated publish workflow using [changesets](https://github.com/atlassian/changesets) that allows CI to automatically deploy new releases - this allows for publishing new icon updates initiated via Dependabot entirely in the GitHub UI, something I've wanted for quite a while
* Support for an <kbd>automerge</kbd> label, that automatically merges PRs when they are mergeable (not a draft, all checks pass, PR is approved, etc.)
* Dependabot updates for GitHub Actions

I may be able to import some of the automated UI testing using Chromatic from @svg-icons/svg-icons repo, but will have to see, the build in this repo is very heavyweight at the moment.

---

Heads-up @stramel, I've moved this repository into its own organization ✨ (you should have a new GitHub invite). I'm not sure if Vercel is following us across the move, I'll have to debug that / see if we need to switch to Netlify.

### CODE REVIEW

**Comment by jacobwgillespie** [comment]
Vercel did not transfer correctly, I've contacted support to see if we can fix that. I tried an experimental Netlify build as well, however Netlify has a 15 minute build time limit, which currently we're exceeding. 🙂

**Comment by jacobwgillespie** [comment]
Update on changes so far...

## Builds

In an attempt to reduce build times, I've switched to building all icon packs in a single sweep as well as generating the JavaScript with Babel, with TypeScript building the `.d.ts` files:

**Before**

* Each icon pack was built individually, using Lerna to run all builds in parallel
* The build script would generate icon source files in a `__build` subdirectory, along with temporary `package.json` files for each icon
* The build script executed TypeScript twice to build the CommonJS and ESModules builds, moving JS files in-between (you can't change the output file extension with TypeScript)
* After building both JavaScript files, the build script would generate a final `package.json` file for each icon (the ones pointing to the built files)

**After**

* Each icon pack generates its source files individually, using `wsrun` to generate in parallel
* The temporary build `package.json` files for each icon are generated along with the final `package.built.json` files
* Three simultaneous builds run over the sources:
  * TypeScript to generate `.d.ts` files
  * Babel using a modern browser preset to generate ESModule files, with the `.esm.js` file extension
  * Babel using `preset-env` to generate the CommonJS files, with the `.js` file extension
* Finally all `package.built.json` files are renamed to `package.json`, replacing the initial build files

The effect of this on GitHub Actions appears to be a reduction of build times from 10 minutes to around 6 minutes, and my MacBook can now build everything in under 5 minutes!

So there are a few upsides to processing the files with Babel:

* Faster ⚡ 
* The Babel styled-components plugin is more advanced than the TypeScript port we've been using
* Build tooling is simpler and therefore more maintainable

There's also two downsides:

* This changes the output JavaScript - I've manually spot-checked that the outputs are compatible.
* This reduces the TypeScript features we can use to whatever is supported by `isolatedModules`, as that is all that the Babel plugin can transform. This is acceptable as we are not doing anything affected by that fact (no `const enum`, no `namespace`).

## Hosting

I attempted to move the website to Netlify, however builds are not fast enough unfortunately. Netlify's build environment is slower than GitHub actions, and it's not able to complete a build of the icons + a Next.js build in under 15 minutes. The 15 minute limit is a hard limit, so unfortunately Netlify is out.

I'll wait to hear from Vercel about setting up our build connection there again. As a plus, Vercel is also very fast. ⚡ 

## Visual Regression Testing

I've added visual regression testing, like we have on @svg-icons/svg-icons, using [Chromatic](https://www.chromatic.com/builds?appId=5efb10e92a9e6b0022ed8597)! 🎉 Chromatic uses GitHub for permissions, so @stramel you should be able to visit that page and approve changes if necessary.

Building the Storybook is slow, so we've lost all the build speed improvements above to the visual regression testing and the builds are back up to 10-15 minutes. But hey, much better PR safety!

---

Side note, @stramel if you visit https://github.com/svg-icons, you have an invite there as well.

**Comment by jacobwgillespie** [comment]
Okay, final changes complete!

## Hosting

Vercel is live again! Thanks to support for sorting out the issues. ❤️ 

## monorepolint

I added [monorepolint](https://monorepolint.com/) which makes it easy to ensure common files like `.gitignore` or details like license fields are correctly set across all packages.

## After Merge

After merge, I'll set the `UI Review` status to required, but it needs a UI Storybook build on the main branch before it'll work for a PR.

---

Switching this from draft to ready-for-review! 🎉

File: tools/builder/clean-package.js [hunk 1]
```diff
@@ -14,11 +17,9 @@ async function run() {
       continue
     }
 
-    console.log(`Removing ${file}`)
+    // console.log(`Removing ${file}`)
```
Comment (stramel) [inline tools/builder/clean-package.js:20]: NIT: Did you mean to leave this in here?

**Comment by stramel** [comment]
LGTM, Awesome work @jacobwgillespie! This is a huge improvement over the previous build process and should prevent some mistakes that we have made in previous releases.

File: tools/builder/clean-package.js (hunk 1 above)
Comment (jacobwgillespie) [reply tools/builder/clean-package.js:20]: Yep, this was intentional - someday I should make it a flag, but the `console.log` is _way_ slower than the delete action it's performing. The primary purpose of the log message was debugging, which is less important now that things are stable, but I left the comment in there for easy un-commenting to debug later.

**Pull request merged**